nsfwOnly.ts

```diff
@@ -0,0 +1,43 @@
+// @ts-nocheck
+/**
+ * @author: @NeoYaBoi
+ * @version: 1.0.0
+ * @description: This plugin checks if the channel is nsfw and responds to user with a specified response if not nsfw 
+ * @license: Null
+ * @example:
+ * ```ts
+ * import { nsfwOnly } from "../plugins/nsfwOnly"; //(change if need be)
+ * import { sernModule, CommandType } from "@sern/handler";
+ * export default commandModule({
+ *    plugins: [ nsfwOnly('response', true/false) ],
+ *    execute: (ctx) => {
+ *       //your code here
+ *    }
+ * })
+ * ```
+ */
+//sern will only try ephemeral if message is a interaction
+ import { type GuildChannel } from "discord.js";
+ import { CommandType, EventPlugin, PluginType } from "@sern/handler";
+ export function nsfwOnly(response: string, ephemeral: boolean): EventPlugin<CommandType.Both> {
+   return {
+     type: PluginType.Event,
+     description: "Checks if the channel is nsfw or not.",
+     async execute(event, controller) {
+       const [ctx] = event;
+       if(ctx.guild == null) {
+        //responds and stops command if not executed in guild (ran in dms).
+         ctx.reply('This command cannot be used here') //Change this if you want
+         return controller.stop()
+       }
+       //checking if channel is nsfw (18+)
+       if(!(ctx.channel! as GuildChannel).nsfw) {

```

Comment:
Create a dedicated function to check if the channel is `NSFW`, this way of checking isn't good
Should be something like
```ts
function isNSFW(channel: Nullish<TextBasedChannel>) {
	// proper checks to see if the channel supports nsfw, if yes then is it enabled or not else return false
}
```

Reply:
I see, defiantly will do but could you explain why its not good?

Reply:
Because you're telling typescript server to assume it's GuildChannel
If in reality its thread channel, it will throw error at runtime or rather just be undefined
It's prone to more errors and makes things harder to debug

Reply:
Ahhh I see. Totally forgot threads were a separate type of channel. Thank you b for clarifying